### PR TITLE
improved robustness of provisioning UI

### PR DESCRIPTION
**Pull request by jfrohnhofen**
* changed styling of the error messages
* added polling and more messages in the UI during network and S3 connection verification
* also added the same "Please wait..." during device formatting
fixes #46

### CODE REVIEW

File: leap-provision-site/src/leap_config.rs [hunk 1]
```diff
@@ -154,8 +160,48 @@ pub fn leap_config_page() -> Html {
 
                 let response = match request.send().await {
                     Ok(r) => r,
-                    Err(e) => {
-                        toast.set(Some(format!("Request failed: {e}")));
+                    Err(_) => {
+                        // Connection dropped — the device is likely switching networks to test
+                        // the S3 configuration (includes NTP sync, which may take extra time).
+                        reconnecting.set(true);
```
Comment (Javier-varez) [inline leap-provision-site/src/leap_config.rs:166]: Unless I missed something, I think this can also trigger due to legitimate errors like:

https://github.com/T4EQ/leap/blob/74baef62fd80599f9386bf36ffb873b0c31b88d7/leap-server/src/provision/cfg.rs#L128

Could we handle actual HTTP responses with error codes differently than connection drops? It should be a better experience for the user to get more specific feedback about the stuff they entered.

File: leap-provision-site/src/leap_config.rs [hunk 2]
```diff
@@ -154,8 +160,48 @@ pub fn leap_config_page() -> Html {
...
+                        for _ in 0..75u32 {
+                            sleep(Duration::from_secs(2)).await;
+                            let Ok(resp) = Request::get("/provision/status").send().await else {
+                                continue;
+                            };
+                            let Ok(status) = resp.json::<ProvisionStatus>().await else {
+                                continue;
+                            };
+                            match status {
+                                ProvisionStatus::LeapConfig => {
```
Comment (Javier-varez) [inline leap-provision-site/src/leap_config.rs]: Maybe let's make https://github.com/T4EQ/leap/blob/74baef62fd80599f9386bf36ffb873b0c31b88d7/leap-api/src/types.rs#L220 derive Eq and then you can just check for equality here.

**Comment by Javier-varez** [comment]
Did you get to test this on a raspberry pi? Or should I try it out?

File: leap-provision-site/src/leap_config.rs (hunk 2 above)
Comment (jfrohnhofen) [reply leap-provision-site/src/leap_config.rs]: Done

File: leap-provision-site/src/leap_config.rs (hunk 1 above)
Comment (jfrohnhofen) [reply leap-provision-site/src/leap_config.rs:166]: Isn't that done in line 210? Or maybe I'm misunderstanding what you mean.

**Comment by jfrohnhofen** [comment]
> Did you get to test this on a raspberry pi? Or should I try it out?

Yes, I ran this on a RPi three times, but did not try to trigger any edge cases. I also did not have a valid S3 configuration, so I never reached the final provisioning step.

File: leap-provision-site/src/leap_config.rs (hunk 1 above)
Comment (Javier-varez) [reply leap-provision-site/src/leap_config.rs:166]: I mean that, now, if there is any error, we get into this loop of up to 75 iterations where we check the status and if the status is still leap configuration that replies with: `LEAP configuration could not be applied...` instead of the actual error code that we got. 

IMO this branch `Err(_) => {` in line 163 is conflating 2 different kind of errors:
- And error in the configuration or request (which would return an HTTP error code different than 200.
- A connectivity error, in which case, getting in this loop makes sense, cause we need to check the status once we reconnect.

And the problem I see is we discard the error text in both cases, while the former could have valuable information (like: e.g.: could not synchronize the clock via ntp)

**Review by Javier-varez — APPROVED** [review]
The remaining comment is more of a suggestion than anything else, LGTM. Also,
to test it with the last step you can use the credentials in vaultwarden.t4eq.org. Use the read-only ones (though read-write would also work, but are not needed)

**Comment by Javier-varez** [comment]
Just tried it on the complete flow. It works well all the way through! Thanks for working on this!

**Comment by jfrohnhofen** [comment]
I'll work on your suggestion and finish this tomorrow

File: leap-provision-site/src/leap_config.rs (hunk 1 above)
Comment (jfrohnhofen) [reply leap-provision-site/src/leap_config.rs:166]: Looking at your comment and the code again i think we are still talking past each other. Here is my understanding: For the initial request (POST to `/provision/config`), if that fails with network-level error (i.e. connection is dropped) `send` fails and we enter the `Err(_)` branch in L163. Otherwise (i.e. if we get a 200 or non-200 response), the `send` succeeds and we go to the `Ok(r) => r` branch. The response status is then checked i L206. If we enter the `Err(_)` branch, because the initial request failed at the network level, we try reaching `provision/status` until we receive a `200` response. And then depending on the `ProvisionStatus` either forward to the next status page or report that leap configuration failed. So the only status not being handled would be a non-200 from the `provision/status` call, but I don't see how that would happen and at that point the server has probably restarted anyways, so any more specific configuration error would have been lost and not be reported by the server through `provision/status`. So I still think the specific configuration error is reported to the user in L206.

What do you think?

Comment (Javier-varez) [reply leap-provision-site/src/leap_config.rs:166]: ah, yes, you are correct, I was assuming the wrong return type when I commented on this, I was under the impression that it would return 400 and 500 errors in the error value of the result.
